api/query/search_test.go

```diff
@@ -43,9 +70,12 @@ func TestLoadSummary_cacheMiss(t *testing.T) {
 
 	// Use channel to synchronize with expected async cache.Put().
 	c := make(chan bool)
-	cache.EXPECT().Get(key).Return(nil, memcache.ErrCacheMiss)
-	store.EXPECT().Get(url).Return(smry, nil)
-	cache.EXPECT().Put(key, smry).DoAndReturn(func(key string, smry []byte) error {
+	w := NewMockWriteCloser(mockCtrl)
```

Comment:
Is this the only reason why we need to mock the whole `io` package?

I believe this, instead of `MockSuccessfulRead`, warrants to be mocked manually as we'd like to make sure `Close` is called properly. And we can delegate to `bytes.Buffer` internally. E.g.

```go
type MockWriteCloser struct {
	buffer bytes.Buffer
	closed bool
	t *testing.T
}

func (mcw *MockWriteCloser) Write(...) {
	assert.False(t, closed)
	buffer.Write(...)
}

func (mcw *MockWriteCloser) Close(...) {
	mcw.closed = true
}

func NewMockWriteCloser(t *testing.T) {
	return MockWriteCloser{
		buffer: bytes.Buffer.NewBuffer(nil)
		closed: false,
		t: t,
	}
}

assert.Equal(t, "expected", mcw.buffer.String())
```